substrate/client/rpc-spec-v2/src/chain_head/chain_head.rs

```diff
@@ -334,16 +357,21 @@ where
 		let mut items = items;
 		items.truncate(num_operations);
 
+		let (rp, rp_is_success) = method_started_response(operation_id, Some(discarded));
+
 		let fut = async move {
+			// Events should only by generated if the response
+			// was successful.
+			if rp_is_success.await.is_err() {
```

Comment:
Just to double check; we should propagate events only if we can propagate the  `method_started_response`, which is always a successful `ResponsePayload::success`?
In other words, the error here means we cannot send the message 🤔 

I would change this to `...if the response was successfully propagated` or something similar